specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2022-04-02-preview/privateLinkScopes.json

```diff
@@ -0,0 +1,1095 @@
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "PrivateEndpointConnection": {
+      "type": "object",
+      "description": "A private endpoint connection",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties",
+          "description": "Resource properties."
+        },
+        "systemData": {
+          "readOnly": true,
+          "description": "The system meta data relating to this resource.",
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/systemData"
+        }
+      }
+    },
+    "PrivateEndpointConnectionListResult": {
+      "type": "object",
+      "description": "A list of private endpoint connections.",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "PublicNetworkAccessType": {
+      "type": "string",
+      "description": "The network access policy to determine if Azure Arc agents can use public Azure Arc service endpoints. Defaults to disabled (access to Azure Arc services only via private link).",
+      "default": "Disabled",
+      "enum": [
+        "Enabled",
+        "Disabled"
+      ],
+      "x-ms-enum": {
+        "name": "PublicNetworkAccessType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Enabled",
+            "description": "Allows Azure Arc agents to communicate with Azure Arc services over both public (internet) and private endpoints."
+          },
+          {
+            "value": "Disabled",
+            "description": "Does not allow Azure Arc agents to communicate with Azure Arc services over public (internet) endpoints. The agents must use the private link."
+          }
+        ]
+      }
+    }
+  },
+  "parameters": {
+    "PrivateLinkScopeName": {
+      "name": "scopeName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the Azure Arc PrivateLinkScope resource.",
+      "x-ms-parameter-location": "method"
+    },
+    "PrivateLinkScopeId": {
+      "name": "privateLinkScopeId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The id (Guid) of the Azure Arc PrivateLinkScope resource.",
+      "x-ms-parameter-location": "method"
+    },
+    "GroupNameParameter": {
+      "name": "groupName",
```

Comment:
>groupName [](http://example.com/codeflow?start=15&length=9)

Why not privateLinkName? What kind of group is it?

Reply:
This is defined in NRP contract: they have to have group per connections
[https://microsoft-my.sharepoint.com/:w:/p/sakulal/Eb17pOEskn1JoKpQM7dpm7sBcxWvnYAoTekqcGf_gUm3og?e=Vvgru8](https://microsoft-my.sharepoint.com/:w:/p/sakulal/Eb17pOEskn1JoKpQM7dpm7sBcxWvnYAoTekqcGf_gUm3og?e=Vvgru8) section#6.3.2